zerver/lib/templates.py

```diff
@@ -146,6 +145,14 @@ def render_markdown_path(
     else:
         extensions = md_extensions
 
+    if help_center:
+        md_macro_extension = zerver.lib.markdown.include.makeExtension(
+            base_path="templates/zerver/help/include/"
+        )
+    else:
+        md_macro_extension = zerver.lib.markdown.include.makeExtension(
+            base_path="templates/zerver/include/markdown_macros/"
```

Comment:
What's the thinking behind `include/markdown_macros` vs say `include/api` and `include/integrations`, or `templates/zerver/markdown_macros/`?

Reply:
Might be a good #documentation discussion as it's kinda a question of "what feels intuitive to others".

Reply:
I wonder if we could make this just read the `api/` part from the path without doing something hideous, so that `templates/zerver/api/` can use `templates/zerver/include/api` without a chain of if/else statements. I suppose the simplest would be passing a `include_namespace="help"` parameter rather than `help_center=True`.

We can always share macros between /api and /integrations via a symlink if there's one that's used in both places.

Reply:
> I wonder if we could make this just read the `api/` part from the path without doing something hideous, so that `templates/zerver/api/` can use `templates/zerver/include/api` without a chain of if/else statements. I suppose the simplest would be passing a `include_namespace="help"` parameter rather than `help_center=True`.

I did think about that, but since this is a prep for moving the help center documentation out of `templates/zerver`, I thought that the boolean was a better option. The follow-up to this will be moving those help center files and the base path for the include would then be completely different.

> What's the thinking behind `include/markdown_macros` vs say `include/api` and `include/integrations`, or `templates/zerver/markdown_macros/`?
>
> Might be a good #documentation discussion as it's kinda a question of "what feels intuitive to others".

I can definitely start that conversation. I can see `templates/zerver/markdown_macros` being a more intuitive option that what I used.

Reply:
Ahh, yeah, I'd forgotten that we were planning to move it further out of `templates/zerver`. One thing on my mind is whether we want something like:

`templates/help/` for the /help/: and `templates/include/help/` for its include content.

That path would kinda signal that we'd have a "help" Django application in Django's file hierarchy; which is probably not a bad impression to signal even if we're not actually doing the bit of overhead to make one.